File: src/dere_plugins/dere_core/scripts/config_reader.py
#!/usr/bin/env python3
"""Helper to read dere configuration."""

from __future__ import annotations

import tomllib

from dere_shared.constants import DEFAULT_ACTIVITYWATCH_URL, DEFAULT_DAEMON_URL
from dere_shared.paths import get_config_dir


def read_config() -> dict:
    """Read dere config.toml."""
    config_path = get_config_dir() / "config.toml"
    if not config_path.exists():
        return {}

    with open(config_path, "rb") as f:
        return tomllib.load(f)


def get_daemon_url() -> str:
    """Get daemon URL from config."""
    config = read_config()
    return config.get("ambient", {}).get("daemon_url", DEFAULT_DAEMON_URL)


def get_activitywatch_url() -> str:
    """Get ActivityWatch URL from config."""
    config = read_config()
    return config.get("activitywatch", {}).get("url", DEFAULT_ACTIVITYWATCH_URL)


if __name__ == "__main__":
    import sys

    if len(sys.argv) < 2:
        print("Usage: config_reader.py <daemon_url|activitywatch_url>")
        sys.exit(1)

    match sys.argv[1]:
        case "daemon_url":
            print(get_daemon_url())
        case "activitywatch_url":
            print(get_activitywatch_url())
        case _:
            print(f"Unknown config key: {sys.argv[1]}", file=sys.stderr)
            sys.exit(1)

File: plugins/dere_core/scripts/config_reader.ts
import { loadConfig } from "@dere/shared-config";
import { DEFAULT_ACTIVITYWATCH_URL, DEFAULT_DAEMON_URL } from "@dere/shared-runtime";

export async function readConfig(): Promise<Record<string, unknown>> {
  return (await loadConfig()) as Record<string, unknown>;
}

export async function getDaemonUrl(): Promise<string> {
  const config = (await loadConfig()) as {
    ambient?: { daemon_url?: string | null };
  };
  return config.ambient?.daemon_url ?? DEFAULT_DAEMON_URL;
}

export async function getActivitywatchUrl(): Promise<string> {
  const config = (await loadConfig()) as {
    activitywatch?: { url?: string | null };
  };
  return config.activitywatch?.url ?? DEFAULT_ACTIVITYWATCH_URL;
}

async function main(): Promise<void> {
  const key = process.argv[2];
  if (!key) {
    console.error("Usage: config_reader.ts <daemon_url|activitywatch_url>");
    process.exit(1);
  }

  switch (key) {
    case "daemon_url": {
      console.log(await getDaemonUrl());
      return;
    }
    case "activitywatch_url": {
      console.log(await getActivitywatchUrl());
      return;
    }
    default: {
      console.error(`Unknown config key: ${key}`);
      process.exit(1);
    }
  }
}

if (import.meta.main) {
  void main();
}
